daemon/logger/factory/factory.go

```diff
@@ -0,0 +1,57 @@
+package factory
+
+import (
+	"fmt"
+	"sync"
+
+	logger "github.com/docker/docker/daemon/logger"
+)
+
+// Creator is a method that builds a logging driver instance with given context
+type Creator func(LoggingContext) (logger.Logger, error)
+
+// LoggingContext provides enough information for a logging driver to do its function
+type LoggingContext struct {
```

Comment:
@LK4D4 when I use type assertion at `container.go` to find out which type of logger was going to be used, I will still need to pass a `path string` to the LoggingContext, which most of the drivers won't use. How is that any better? Also wouldn't this require moving the `"-json"` string back to the daemon pkg? Should daemon continue to be aware of specific types of loggers?

I don't think this is making all drivers bad. It's just an method that's inapplicable to some drivers, which they can discard by not using this field.